Enable the pylint-django migrations plugin

https://github.com/pylint-dev/pylint-django?tab=readme-ov-file#additional-plugins

Today we're not actually running pylint against any of the migration files. If we enable pylint for these files we encounter some pylint rules that we probably don't want to deal with in migrations:

```
Using config file pyproject.toml
************* Module nautobot_dev_example.migrations.0001_initial
nautobot_dev_example/migrations/0001_initial.py:1:0: C0114: Missing module docstring (missing-module-docstring)
nautobot_dev_example/migrations/0001_initial.py:1:0: C0103: Module name "0001_initial" doesn't conform to snake_case naming style (invalid-name)
nautobot_dev_example/migrations/0001_initial.py:11:0: C0115: Missing class docstring (missing-class-docstring)
************* Module nautobot_dev_example.migrations.0002_runpython
nautobot_dev_example/migrations/0002_runpython.py:1:0: C0114: Missing module docstring (missing-module-docstring)
nautobot_dev_example/migrations/0002_runpython.py:1:0: C0103: Module name "0002_runpython" doesn't conform to snake_case naming style (invalid-name)
nautobot_dev_example/migrations/0002_runpython.py:6:0: C0116: Missing function or method docstring (missing-function-docstring)
nautobot_dev_example/migrations/0002_runpython.py:10:0: C0115: Missing class docstring (missing-class-docstring)
nautobot_dev_example/migrations/0002_runpython.py:15:18: W5197: Always include backwards migration callable (missing-backwards-migration-callable)
```

As far as I can tell the global pylint config in the pyproject.toml allows you to either:
- ignore entire files or directories
- disable rules globally

But it does not allow you to disable rules for specific files or directories. Our only option may be to run pylint twice, the second time only checking migrations and disabling some rules:

```sh
pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml nautobot_dev_example
pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml --module-rgx='^[0-9a-z_]{2,30}$' --disable=missing-module-docstring,missing-class-docstring,missing-function-docstring nautobot_dev_example.migrations
```
```

--- a/changes/183.added
+++ b/changes/183.added
@@ -1,0 +1,1 @@
+Added pylint django migrations checker to the `invoke pylint` command.

--- a/nautobot-app/{{ cookiecutter.project_slug }}/tasks.py
+++ b/nautobot-app/{{ cookiecutter.project_slug }}/tasks.py
@@ -710,21 +710,24 @@ def hadolint(context):
 def pylint(context):
     """Run pylint code analysis."""
     exit_code = 0
-    run_command(context, command)
 
     base_pylint_command = 'pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml'
     command = f"{base_pylint_command} {{ cookiecutter.app_name }}"
     if not run_command(context, command, warn=True):
         exit_code = 1
 
-    # run the pylint_django migrations checkers on the migrations directory
-    migrations_pylint_command = (
-        f"{base_pylint_command} --load-plugins=pylint_django.checkers.migrations"
-        " --disable=all --enable=new-db-field-with-default,missing-backwards-migration-callable"
-        " {{ cookiecutter.app_name }}.migrations"
-    )
-    if not run_command(context, migrations_pylint_command, warn=True):
-        exit_code = 1
+    # run the pylint_django migrations checkers on the migrations directory, if one exists
+    migrations_dir = Path(__file__).absolute().parent / Path("{{ cookiecutter.app_name }}") / Path("migrations")
+    if migrations_dir.is_dir():
+        migrations_pylint_command = (
+            f"{base_pylint_command} --load-plugins=pylint_django.checkers.migrations"
+            " --disable=all --enable=fatal,new-db-field-with-default,missing-backwards-migration-callable"
+            " {{ cookiecutter.app_name }}.migrations"
+        )
+        if not run_command(context, migrations_pylint_command, warn=True):
+            exit_code = 1
+    else:
+        print("No migrations directory found, skipping migrations checks.")
 
     raise Exit(code=exit_code)
 

--- a/nautobot-app/{{ cookiecutter.project_slug }}/tasks.py
+++ b/nautobot-app/{{ cookiecutter.project_slug }}/tasks.py
@@ -709,8 +709,24 @@ def hadolint(context):
 @task
 def pylint(context):
     """Run pylint code analysis."""
-    command = 'pylint --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml {{ cookiecutter.app_name }}'
+    exit_code = 0
     run_command(context, command)
+
+    base_pylint_command = 'pylint --verbose --init-hook "import nautobot; nautobot.setup()" --rcfile pyproject.toml'
+    command = f"{base_pylint_command} {{ cookiecutter.app_name }}"
+    if not run_command(context, command, warn=True):
+        exit_code = 1
+
+    # run the pylint_django migrations checkers on the migrations directory
+    migrations_pylint_command = (
+        f"{base_pylint_command} --load-plugins=pylint_django.checkers.migrations"
+        " --disable=all --enable=new-db-field-with-default,missing-backwards-migration-callable"
+        " {{ cookiecutter.app_name }}.migrations"
+    )
+    if not run_command(context, migrations_pylint_command, warn=True):
+        exit_code = 1
+
+    raise Exit(code=exit_code)
 
 
 @task(aliases=("a",))

